Add: Optional static analysis during build

mpconley opened the pull request:
#### Brief overview of PR changes/additions

Adds optional static analysis integration to the CMake build system. Developers can now enable clang-tidy and cppcheck during compilation by passing `-DENABLE_STATIC_ANALYSIS=ON` to cmake. The analysis runs automatically during builds and helps catch bugs, performance issues, and code quality problems.

**Changes:**
- New `cmake/StaticAnalysis.cmake` module with clang-tidy and cppcheck configuration
- Enabled by default in `CMakeLists.txt` but requires opt-in flag to activate
- Updated AI assistant instructions with wiki reference
- Performs performance-*, bugprone-*, and clang-analyzer-* checks

#### Motivation for adding to Mudlet

Static analysis helps developers catch issues early without requiring manual code review:
- **Bug detection**: Identifies null pointer dereferences, memory leaks, and logic errors
- **Performance**: Finds unnecessary copies and inefficient patterns
- **Code quality**: Enforces best practices and consistent patterns
- **Developer experience**: Optional feature - doesn't impact regular builds

Mudlet already uses static analysis in CI, but this makes it available locally for faster feedback during development.

#### Other info (issues closed, discussion etc)

- Static analysis is **disabled by default** - developers opt-in with the cmake flag
- Build output can be captured and filtered to focus on Mudlet-specific issues
- Uses existing `.clang-tidy` configuration already in the repository
- Wiki documentation will be added at https://wiki.mudlet.org/w/Compiling_Mudlet#Static_Analysis

mpconley committed:
Add CMake static analysis integration with clang-tidy

- Add cmake/StaticAnalysis.cmake with clang-tidy and cppcheck configuration
- Add commented include(StaticAnalysis) in root CMakeLists.txt
- Update docs/ai-instructions.md with static analysis wiki reference

To enable static analysis, uncomment the include(StaticAnalysis) line
and reconfigure with -DENABLE_STATIC_ANALYSIS=ON

mpconley committed:
Fix StaticAnalysis.cmake: Remove broken --analyze flag

The --analyze flag switches Clang into analysis-only mode and prevents
object file generation, breaking the build.

Changes:
- Remove ENABLE_CLANG_ANALYZER option and --analyze flags
- Clang Static Analyzer already runs via clang-tidy's clang-analyzer-* checks
- Add comprehensive documentation and usage instructions
- Improve status messages to show enabled check categories
- Make cppcheck optional (info message instead of warning if missing)

Static analysis now works correctly during normal builds without
breaking compilation or linking.

mpconley committed:
Enable StaticAnalysis.cmake by default (opt-in via flag)

Uncomment include(StaticAnalysis) - the module is safe to include because
it defaults ENABLE_STATIC_ANALYSIS to OFF. Developers opt-in by passing
-DENABLE_STATIC_ANALYSIS=ON to cmake.

This is the standard CMake pattern - optional features controlled by flags,
not by commenting/uncommenting includes.

vadi2 reviewed (commented):
Looks awesome. Would you be able to add [clazy](https://github.com/KDE/clazy) to the mix as well? It is a Qt best practices static analyzer.

mpconley committed:
Add: macOS build instructions and clazy documentation

- Add macOS-specific make command with sysctl -n hw.ncpu
- Document clazy Qt-specific static analysis as a separate build mode
- Include installation and usage instructions for clazy

mpconley commented:
> Looks awesome. Would you be able to add [clazy](https://github.com/KDE/clazy) to the mix as well? It is a Qt best practices static analyzer.

Thanks for the suggestion. I investigated clazy integration and found that it works differently from clang-tidy - it requires using clazy as the compiler (CXX=clazy) rather than running alongside the build like clang-tidy does via CMAKE_CXX_CLANG_TIDY.

I've added documentation for running clazy as a separate analysis mode:
```
# Clean build directory required
CLAZY_CHECKS="level0,level1" CXX=clazy cmake ..
make -j `sysctl -n hw.ncpu`
```
This keeps the main static analysis feature simple (clang-tidy + cppcheck during regular builds) while providing guidance for Qt-specific analysis when needed.

mpconley committed:
Fix: Clarify static analysis comment to match implementation

The StaticAnalysis module is always included but only activates when
-DENABLE_STATIC_ANALYSIS=ON is passed. Updated comment to avoid
confusion about unconditional include vs. configurable behavior.
